### ISSUE TITLE
PEcAn.utils: Remove GDAL dependency by moving regrid and grid2netcdf

### ISSUE THREAD

**Issue by infotroph** [PecanProject/pecan#2913]
Pasting from Slack discussion:

[infotroph](https://app.slack.com/team/U8TF1TK8X) [Today at 11:51](https://pecanproject.slack.com/archives/G8S9VG292/p1645527068356649)
Does anyone actively use the PEcAn.utils functions regrid or grid2netcdf? They seem unused inside PEcAn and are the only functions in utils that use raster,  which in turn needs the GDAL libraries, so removing them would eliminate a major build dependency for all packages that don’t need to use spatial processing. (edited) 
5 replies

[infotroph](https://app.slack.com/team/U8TF1TK8X)  [5 hours ago](https://pecanproject.slack.com/archives/G8S9VG292/p1645527292268479?thread_ts=1645527068.356649&cid=G8S9VG292)
(I noticed this because I managed to break my local GDAL in a way that made R segfault on make .check/base/utils :grimacing:… but that probably says more about how badly I broke GDAL than about PEcAn)

[Shawn](https://app.slack.com/team/U8QSSVA7K)  [3 hours ago](https://pecanproject.slack.com/archives/G8S9VG292/p1645534135422809?thread_ts=1645527068.356649&cid=G8S9VG292)
[@infotroph](https://pecanproject.slack.com/team/U8TF1TK8X) we could move that functionality to data.remote since its perhaps more related to the geospatial functions there which already also has raster as a depends.

[Dietze](https://app.slack.com/team/U8RBH1467)  [3 hours ago](https://pecanproject.slack.com/archives/G8S9VG292/p1645534140201399?thread_ts=1645527068.356649&cid=G8S9VG292)
Right now they are only used in [base/visualization/vignettes/usmap.Rmd](https://github.com/PecanProject/pecan/blob/ee38544e1c995f47073019bd31b3179e7e64991a/base/visualization/vignettes/usmap.Rmd)  All of this seems to be [@dlebauer](https://pecanproject.slack.com/team/U8S7AMWAX). The other thing that's intuitive about these functions is that they are hard-coded to just write out Yield data, so they'd need generalization if we wanted to put them in CRAN.The code does seem to be handy, so I'm hesitant to delete completely, but wonder if we could more to another package that is already using raster (e.g. data.remote?) or to an /inst folder for preserving until we have the bandwidth to generalize.

[Shawn](https://app.slack.com/team/U8QSSVA7K)  [3 hours ago](https://pecanproject.slack.com/archives/G8S9VG292/p1645534189601449?thread_ts=1645527068.356649&cid=G8S9VG292)
hahahaha!  [@Dietze](https://pecanproject.slack.com/team/U8RBH1467) we had that same thought at about the same time.  Thats two votes for data.remote.

[Shawn](https://app.slack.com/team/U8QSSVA7K)  [3 hours ago](https://pecanproject.slack.com/archives/G8S9VG292/p1645534395222429?thread_ts=1645527068.356649&cid=G8S9VG292)
grid2netcdf doesnt look too bad to generalize either.  Mostly remote hard coded cropyield variable name and a little other tweaking so that we can save other datasets as netCDF where appropriate

**Comment by dlebauer** [comment]
I think that code is c 2014 … I no longer need it but definitely move if it is / would be useful. I think it was necessary to plot output generated on the NARR grid

**Comment by meetagrawal09** [comment]
@infotroph , should we consider moving `regrid` and `grid2netcdf` to `data.remote` package to solve this one ?

**Comment by meetagrawal09** [comment]
What would be the best way to manage dependency on the `regrid` function here : https://github.com/PecanProject/pecan/blob/develop/base/visualization/vignettes/usmap.Rmd

**Comment by mdietze** [comment]
I'd just delete the example in the vignette

```{r fig.width=7, fig.height=6}
latlon.data <- read.csv("~/dev/bety/public/temp_models/willow_yield_grid.csv",
                        skip = 1)
regridded <- regrid(latlon.data)
library(ggplot2)
ggplot(data = regridded) + geom_tile(aes(lat, lon, fill = yield))
```

it's not going to work anyways since it depends on a hard path to a file that doesn't exist on the user's computer

**Comment by meetagrawal09** [comment]
And an exactly same approach for this one https://github.com/PecanProject/pecan/blob/develop/base/visualization/vignettes/usmap.Rmd#L95-#L125 ?

**Comment by mdietze** [comment]
Yeah, we don't need a vignette on how to process old biocro output
